Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs

```diff
@@ -174,8 +174,16 @@ private void InitializeServiceLocator()
                 Utilities.Editor.InputMappingAxisUtility.CheckUnityInputManagerMappings(Definitions.Devices.ControllerMappingLibrary.UnityInputManagerAxes);
 #endif
 
-                if (!RegisterService(typeof(IMixedRealityInputSystem), Activator.CreateInstance(ActiveProfile.InputSystemType) as IMixedRealityInputSystem) ||
-                    InputSystem == null)
+                bool registrationSucceeded = false;
+                try
+                {
+                    registrationSucceeded = RegisterService(typeof(IMixedRealityInputSystem), Activator.CreateInstance(ActiveProfile.InputSystemType) as IMixedRealityInputSystem);
+                }
+                catch (Exception e)
+                {
+                    DebugUtilities.LogCriticalError($"Registering the Input System threw: {e.GetType().ToString()} - {e.Message}.");
```

Comment:
Should we add `DebugUtilities.LogCriticalError(string message, Exception e)` to have common formatting?

Reply:
maybe. Not a bad thought